### PR TITLE
report metrics to influxdb

### PR DESCRIPTION
**Pull request by wangtuanjie**
#90

### CODE REVIEW

**Comment by yichengq** [comment]
The basic package to report to influxdb looks great. But we need to discuss more about how and what to report:
1. I see you separate client_ops and server_ops, but i prefer to treat them as one event, which has attributes 'clientID', 'disk', 'op'. Users could get more info from it, e.g., the op distribution from some client.
2. The value in metrics is also used in enforce pkg today: https://github.com/c-fs/cfs/blob/master/enforce/enforce.go , so you cannot simply remove it. It is used to judge whether the number of ops in a period of time exceeds the quota.

Have you played it with real influxdb?

**Comment by wangtuanjie** [comment]
1. Multi measurements want reduce points. If all in one measurement which must have more tags, this will lead  slowly query. We should decide a balance between them.
2. I forget the enforce package and stats api  depend on metric package. I think we can use a ratelimit such as https://github.com/juju/ratelimit to implement quota. I don't have mind on stats api. How do you think about it?
3. Have you played it with real influxdb? Sure, graph like below
   qq20150731-2 2x

**Comment by yichengq** [comment]
1. ok. I am fine with your choice considering it has fulfilled my expectation.
2. My thought is that there is only one report place, and all components(stats, enforce, influxdb) could get data from it. Only one input could make code quite clean. Today, the report place is at stats package, and the original report data is saved at codahale/metrics. All components could get data from codahale/metrics, but they don't make changes on it. For influxdb, i think it could keep a latest snapshot and send the diff into the database.
3. Awesome graph! :+1:

**Comment by wangtuanjie** [comment]
@yichengq Ready for review.

File: stats/stats.go [hunk 1]
```diff
@@ -34,28 +37,26 @@ func SetContainerName(name string) { containerName = name }
...
-	var prefix string
-	if c.disk != "" {
-		prefix = c.disk + "_"
+func (c *CounterType) Add() {
+	metrics.Counter(c.disk + "_" + c.op).Add()
+	if c.client != 0 {
+		metrics.Counter(clientCounterPrefix + strconv.FormatInt(c.client, 16)).Add()
 	}
-	metrics.Counter(prefix + c.name).Add()
-	metrics.Counter(strconv.FormatInt(c.client, 16) + "_ops").Add()
```
Comment (yichengq) [inline stats/stats.go:58]: enforce pkg is relying on this format now: https://github.com/c-fs/cfs/blob/master/enforce/enforce.go#L51
Could we make them consistent?

File: stats/container.go [hunk 2]
```diff
@@ -29,7 +26,7 @@ func initContainerManager() {
 		return
 	}
 	// TODO: support influxdb or other backend storage
-	cmgr, err = manager.New(memory.New(storageDuration, nil), sysFs, maxHousekeepingInterval, allowDynamicHousekeeping)
+	cmgr, err = manager.New(memory.New(storageDuration, nil), sysFs)
```
Comment (yichengq) [inline stats/container.go]: could you update to latest github.com/google/cadvisor ? then the original code will work.

Comment (wangtuanjie) [inline stats/container.go]: Ack.

File: stats/stats.go (hunk 1 above)
Comment (wangtuanjie) [inline stats/stats.go:58]: Ack.

**Comment by yichengq** [comment]
LGTM!
The functions are pretty clean and readable. And the feature is wonderful~~
Thanks for making this happen!

There is still some pieces left in enforce side and I will send PR for it.

**Pull request merged**